### ISSUE TITLE
Does the metadata include compiler version and execution platform information

### ISSUE THREAD

**Issue by doomsdayciel** [IBM/Project_CodeNet#31]
Hi,

Thank you for sharing this wonderful projects.
You've annotated a rich set of information in metadata, but does the metadata include compiler version and execution platform information.

I think the compiler version or execution platform(ex.x86 ) may impact the execute result, and some programming syntax.

So how can we know the compiler version and platform information for the code sample in the dataset?
If these information were not included now, is it possible to add them?

Thank you!

**Comment by geert56** [comment]
Yes, apart from the programming language, the metadata for most submissions also indicate a more precise description of the required language/compiler version. Please consult one of the .csv metadata files and look under the "original_language" column. For instance you will find entries such as "C++ (GCC 9.2.1)". However there is no prescribed format for this field and it is up to user to parse it. It is not known what the possible values are and their meaning can only be guessed. It is what was originally reported to the judging site. We only consolidated the "language" column.

**Comment by doomsdayciel** [comment]
> Yes, apart from the programming language, the metadata for most submissions also indicate a more precise description of the required language/compiler version. Please consult one of the .csv metadata files and look under the "original_language" column. For instance you will find entries such as "C++ (GCC 9.2.1)". However there is no prescribed format for this field and it is up to user to parse it. It is not known what the possible values are and their meaning can only be guessed. It is what was originally reported to the judging site. We only consolidated the "language" column.

Thanks for the reply, so all the data in the metadata csv is from the judging site? Will Codenet project execute and verify the code sample information before adding into the metadata?

**Comment by ruchirpuri** [comment]
The metadata is curated from the data in judging sites.. We also provide some important tools to make it easy to process the code samples for several languages for which data is copious (C/C++/Python/Java), e.g., for generating Parse Trees, Project_CodeNet/tools/spt-generator/  . We had already filtered some code samples to make the data high quality for CodeNet and will continue to work more on this dimension.
